src/glfw3.jl

```diff
@@ -389,17 +389,11 @@ Base.show(io::IO, vm::VidMode) = write(io, "$(vm.width)x$(vm.height)@$(vm.refres
 struct GLFWImage
 	width::Cint
 	height::Cint
-	pixels::Ptr{Cuchar}
+	pixels::Ptr{UInt8}
 end
 
-function Base.cconvert(::Type{GLFWImage}, image::Matrix{NTuple{4, UInt8}})
-	ptr = Base.cconvert(Ptr{UInt8}, image)
-	(size(image)..., ptr)
-end
-
-function Base.unsafe_convert(::Type{GLFWImage}, data::Tuple{Int, Int, Matrix{NTuple{4, UInt8}}})
-	ptr = Base.unsafe_convert(Ptr{UInt8}, data[3])
-	GLFWImage(Cint(data[1]), Cint(data[2]), ptr)
+function Base.convert(::Type{GLFWImage}, image::AbstractMatrix{NTuple{4,UInt8}})
+	GLFWImage(size(image)...,  Base.unsafe_convert(Ptr{UInt8}, image))
```

Comment:
I think there may be an exceptional case that the GC can erase `image`,  and thus this reference would then be illegal.

Reply:
yeah that looks wrong... I'm pretty sure you need to cascade cconvert & unsafe_convert in some way, to make ccall root image! maybe not like i did, but i think you should at least call unsafe_convert only on arguments to cconvert!